### PR TITLE
feat: force flush LSN opcode for migration finalization

### PR DESCRIPTION
**Pull request by BorysTheDev**
fixes: https://github.com/dragonflydb/dragonfly/issues/4329

### CODE REVIEW

File: src/server/journal/streamer.cc [hunk 1]
```diff
@@ -230,14 +230,19 @@ void RestoreStreamer::Run() {
...
+  journal::Entry noop(journal::Op::NOOP, 0);
 
   io::StringSink sink;
   JournalWriter writer{&sink};
   writer.Write(entry);
+
+  // We do flush sending NOOP
+  for (int i = 0; i < 16; ++i) {
+    writer.Write(noop);
+  }
```
Comment (chakaz) [inline src/server/journal/streamer.cc:242]: Is this really the best way to flush a stream? :upside_down_face:

Comment (BorysTheDev) [reply src/server/journal/streamer.cc:242]: we do near the same in replication

**Comment by adiholden** [comment]
@BorysTheDev where you able to see a change with this pr?
i.e if you run without it sometimes you see restart of migration finalization and after this change you dont see it?

**Comment by BorysTheDev** [comment]
> @BorysTheDev where you able to see a change with this pr? i.e if you run without it sometimes you see restart of migration finalization and after this change you dont see it?

I try to reproduce this situation

**Comment by BorysTheDev** [comment]
I can not reproduce the issue, so I will close the PR

**Pull request closed without merging**